### ISSUE TITLE
Show "Publish Reviews" Button When Unpublished Reviews Exist

### ISSUE THREAD

**Issue by jjeffryes** [OpenBazaar/openbazaar-desktop#1508]
Currently, new reviews for a listing aren't published until the user makes some other change to public information.

We should show a button to publish new reviews somewhere in the UI when unpublished reviews exist.

**Comment by rmisio** [comment]
@jjeffryes Do you mean when a buyer leaves a review, that review is not published unless they change some other "publish triggering" data?

If so, then my question would be why doesn't a new review trigger a publish? I understand we want to limit over publishing on frequent not critical actions, like following... but a review seems pretty infrequent to me and pretty important.

**Comment by jjeffryes** [comment]
Yes, that's exactly it.

I think the idea was publishing can only be triggered by the node owner, not by actions taken by other people?

**Comment by tyler-smith** [comment]
The vendor's node would need to do the publishing, but there's no reason the server shouldn't be able to periodically check for new reviews and publish.

**Comment by rmisio** [comment]
Yeah, that's what I was going to suggest... have some process that is checking and if it sees more than X unpublished reviews or a single unpublished review older than Y, it publishes.

**Comment by SamPatt** [comment]
The original rationale for new reviews not triggering a publish was that it renders the existing reseeded store / listings outdated, and we didn't want that to be able to be triggered by external actions. I think we thought it could have been abused.

If it looks like that's not possible in practice then it can be changed.

**Comment by rmisio** [comment]
> The original rationale for new reviews not triggering a publish was that it renders the existing reseeded store / listings outdated, and we didn't want that to be able to be triggered by external actions. I think we thought it could have been abused.

Well, but this would require a purchase of a listing... right? If that's so, I think most every vendor would trade an outdated IPFS hash for a sale.

I thought the original rational may have been similar to the following rational where we don't want to publish and break caching if potentially dozens of reviews are coming in, in a small window.

If that's true, then I think this fixes things:

```
have some process that is checking and if it sees more than X unpublished reviews or a single unpublished review older than Y, it publishes.
```
